File: frontend/lib/model/Task.dart
class Task {
   String id;
   String title;
   String? description;
   DateTime due;
   bool isDone;

  Task({ required this.id,required this.title,this.description, required this.due, required this.isDone,});

  factory Task.fromJson(Map<String, dynamic> json) => Task(
      id: json['_id'],
      title: json['title'],
      description: json['description']??"",
      due: DateTime.parse(json['due']),
      isDone: json['isDone'],
    );
}

File: backend/src/models/task.js
import { Schema, Types, model } from "mongoose";

const taskSchema = new Schema(
  {
    title: {
      type: String,
      required: true,
    },
    description: {
      type: String,
    },
    due: {
      type: Date,
      required: true,
    },
    user: {
      type: Types.ObjectId,
      required: true,
    },
    isDone: {
      type: Boolean,
      default: false,
    },
  },
  { timestamps: true }
);

export const Task = model("Task", taskSchema);
